src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -2989,47 +3097,100 @@ tx_gateway_frontend::get_all_transactions() {
               vlog(
                 txlog.error,
                 "can't get tm stm of the {}' partition",
-                model::legacy_tm_ntp);
-              co_return tx_errc::unknown_server_error;
+                tx_manager_ntp);
+              return ss::make_ready_future<return_all_txs_res>(
+                return_all_txs_res{tx_errc::unknown_server_error});
           }
 
           auto gate_lock = gate_guard(self._gate);
-          auto read_lock = co_await stm->read_lock();
-
-          auto res = co_await stm->get_all_transactions();
-          if (!res.has_value()) {
-              if (res.error() == tm_stm::op_status::not_leader) {
-                  co_return tx_errc::not_coordinator;
-              }
-              co_return tx_errc::unknown_server_error;
-          }
+          return stm->read_lock().then([stm](ss::basic_rwlock<>::holder unit) {
+              return stm->get_all_transactions()
+                .then(
+                  [](tm_stm::get_txs_result res)
+                    -> ss::future<return_all_txs_res> {
+                      if (!res.has_value()) {
+                          if (res.error() == tm_stm::op_status::not_leader) {
+                              return ss::make_ready_future<return_all_txs_res>(
+                                return_all_txs_res{tx_errc::not_coordinator});
+                          }
+                          return ss::make_ready_future<return_all_txs_res>(
+                            return_all_txs_res{tx_errc::unknown_server_error});
+                      }
+                      return ss::make_ready_future<return_all_txs_res>(
+                        res.value());
+                  })
+                .finally([u = std::move(unit)] {});
+          });
+      });
+}
 
-          co_return res.value();
+ss::future<tx_gateway_frontend::return_all_txs_res>
+tx_gateway_frontend::get_all_transactions() {
+    auto ntp_meta = _metadata_cache.local().get_topic_metadata(
+      model::tx_manager_nt);
+    if (!ntp_meta) {
+        vlog(
+          txlog.error,
+          "Topic {} doesn't exist in metadata cache",
+          model::tx_manager_nt);
+        co_return tx_errc::partition_not_exists;
+    }
+    tx_gateway_frontend::return_all_txs_res res{{}};
+    co_await ss::do_for_each(
```

Comment:
use a normal for-loop: you are in a coroutine.

Reply:
fixed